external/header.php

```diff
@@ -112,17 +128,17 @@ public static function __callstatic($name, $arguments)
 unset($domain);
 
 //Display warning if extension not available
-if (extension_loaded('xhprof') && $_xhprof['doprofile'] === true) {
+if (( $_xhprof['profiler'] ) && $_xhprof['doprofile'] === true) {
     include_once dirname(__FILE__) . '/../xhprof_lib/utils/xhprof_lib.php';
     include_once dirname(__FILE__) . '/../xhprof_lib/utils/xhprof_runs.php';
     if (isset($ignoredFunctions) && is_array($ignoredFunctions) && !empty($ignoredFunctions)) {
-        xhprof_enable(XHPROF_FLAGS_CPU + XHPROF_FLAGS_MEMORY, array('ignored_functions' => $ignoredFunctions));
+        call_user_func($_xhprof['profiler']."_enable", constant(strtoupper($_xhprof['profiler'])."_TIDEWAYS_FLAGS_CPU" ) + constant(strtoupper($_xhrprof['profiler'])."_FLAGS_MEMORY"), array('ignored_functions' => $ignoredFunctions));
     } else {
-        xhprof_enable(XHPROF_FLAGS_CPU + XHPROF_FLAGS_MEMORY);
+        call_user_func($_xhprof['profiler']."_enable", constant(strtoupper($_xhprof['profiler'])."_FLAGS_CPU") + constant(strtoupper($_xhprof['profiler'])."_FLAGS_MEMORY") );
```

Comment:
Please extract `constant(strtoupper($_xhprof['profiler'])."_FLAGS_CPU") + constant(strtoupper($_xhprof['profiler'])."_FLAGS_MEMORY")` into temp variable, because it's used twice and each usage got harder to read, when extension name is configurable.